source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -128,180 +127,119 @@ MPxNode::SchedulingType MaterialXNode::schedulingType() const
+    {
+        dataHandle.set(_xmlElementPath);
+    }
+    else
+    {
+        return MPxNode::getInternalValue(plug, dataHandle);
+    }
 
-	if (plug == DOCUMENT_ATTRIBUTE)
-	{
-		//MString documentContent = dataHandle.asString();
-		//materialXData->createDocument(documentContent.asChar());
-	}
-	else if (plug == ELEMENT_ATTRIBUTE)
-	{
-		if (materialXData->getDocument())
-		{
-			//MString elementPath = dataHandle.asString();
-			//materialXData->element = materialXData->doc->getDescendant(elementPath.asChar());
-		}
-	}
-	else if (plug == _outAttr)
-	{
-		return MPxNode::setInternalValue(plug, dataHandle);
-	}
-	else
-	{
-		std::string plugName = plug.partialName().asChar();
-		if (plug.isChild())
-		{
-			MPlug plug2 = plug.parent();
-			plugName = plug2.partialName().asChar();
-		}
-		if (_attributeElementPairMap.count(plugName) > 0)
-		{
-			MaterialX::ElementPtr element = _attributeElementPairMap[plugName];
-			if (element->isA<MaterialX::ValueElement>())
-			{
-				MaterialX::ValueElementPtr valueElement = element->asA<MaterialX::ValueElement>();
-				std::string type = valueElement->getType();
-				if (type == MaterialX::TypedValue<MaterialX::Vector2>::TYPE)
-				{
-                    float2& value = dataHandle.asFloat2();
-                    valueElement->setValue(MaterialX::Vector2(value[0], value[1]));
-				}
-				else if (type == MaterialX::TypedValue<MaterialX::Vector3>::TYPE)
-				{
-                    float3& value = dataHandle.asFloat3();
-                    valueElement->setValue(MaterialX::Vector3(value[0], value[1], value[2]));
-				}
-				else if (type == MaterialX::TypedValue<MaterialX::Vector4>::TYPE)
-				{
-                    MFloatVector& value = dataHandle.asFloatVector();
-                    valueElement->setValue(MaterialX::Vector4(value[0], value[1], value[2], value[3]));
-				}
-				else if (type == MaterialX::TypedValue<MaterialX::Color2>::TYPE)
-				{
-                    float2& value = dataHandle.asFloat2();
-                    valueElement->setValue(MaterialX::Color2(value[0], value[1]));
-				}
-				else if (type == MaterialX::TypedValue<MaterialX::Color3>::TYPE)
-				{
-					float3& value = dataHandle.asFloat3();
-					valueElement->setValue(MaterialX::Color3(value[0], value[1], value[2]));
-				}
-				else if (type == MaterialX::TypedValue<MaterialX::Color4>::TYPE)
-				{
-					MFloatVector& value = dataHandle.asFloatVector();
-                    valueElement->setValue(MaterialX::Color4(value[0], value[1], value[2], value[3]));
-				}
-				else if (type == MaterialX::TypedValue<float>::TYPE)
-				{
-					float& value = dataHandle.asFloat();
-					valueElement->setValue(value);
-				}
-			}
-		}
-	}
-	return true;
+    return true;
 }
 
-void MaterialXNode::setAttributeValue(MObject &materialXObject, MObject &attr, float* values, unsigned int size, MDGModifier& mdgModifier)
+bool MaterialXNode::setInternalValue(const MPlug& plug, const MDataHandle& dataHandle)
```

Comment:
This method gives us the opportunity to perform the necessary updates on attribute value changes.